CMakeLists.txt

```diff
@@ -121,22 +121,34 @@ else()
 endif()
 
 #---Try to download a file to check internet connection-----------------------------------------
-message(STATUS "Checking internet connectivity...")
-file(DOWNLOAD https://root.cern/files/cmake_connectivity_test.txt ${CMAKE_CURRENT_BINARY_DIR}/cmake_connectivity_test.txt
-  TIMEOUT 10 STATUS DOWNLOAD_STATUS
-)
-# Get the status code from the download status
-list(GET DOWNLOAD_STATUS 0 STATUS_CODE)
-# Check if download was successful.
-if(${STATUS_CODE} EQUAL 0)
-  # Succcess
-  message(STATUS "Yes")
-  # Now let's delete the file
-  file(REMOVE ${CMAKE_CURRENT_BINARY_DIR}/cmake_connectivity_test.txt)
-  set(NO_CONNECTION FALSE)
+if(check_connectivity)
```

Comment:
Can we not use `if(NOT NO_CONNECTION)` or `if(NOT DEFINED NO_CONNECTION)` or `if(NOT CACHE NO_CONNECTION)`? (I don't know which one, not expert enough...)

Reply:
Not sure I understand what you mean. `check_connectivity` is a new `ROOT_BUILD_OPTION`, and `NO_CONNECTION` was already there. Not sure about the `CACHE` option either...

Reply:
I am wondering whether we can directly use `NO_CONNECTION` as ROOT option, instead of introducing an additional `check_connectivity`.

Reply:
I don't think so. But let's discuss this when you're back